core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -600,8 +611,16 @@ private[spark] object JsonProtocolSuite extends Assertions {
             assert(stageAttemptId1 === stageAttemptId2)
             assertSeqEquals[AccumulableInfo](updates1, updates2, (a, b) => a.equals(b))
           })
-        assertOptionEquals(e1.executorUpdates, e2.executorUpdates,
-        (e1: ExecutorMetrics, e2: ExecutorMetrics) => assertEquals(e1, e2))
+        assertSeqEquals[((Int, Int), ExecutorMetrics)](
+          e1.executorUpdates.toSeq.sortWith((x, y) => lexOrder(x._1, y._1)),
+          e2.executorUpdates.toSeq.sortWith((x, y) => lexOrder(x._1, y._1)),
+          (a, b) => {
+            val (k1, v1) = a
+            val (k2, v2) = b
+            assert(k1 === k2)
+            assertEquals(v1, v2)
+          }
```

Comment:
can use pattern matching earlier to clean up a bit more:

```scala
          { case ((k1, v1), (k2, v2)) =>
            assert(k1 === k2)
            assertEquals(v1, v2)
          }
```

Reply:
Yes, this is marginally shorter, but I'd leave it as is to keep it consistent with the immediately preceding `assertSeqEquals` (which uses the same way of destructuring the objects to be compared).